lib/main.go

```diff
@@ -200,6 +200,41 @@ func (obj *Main) Exit(err error) {
 	obj.exit <- err // trigger an exit!
 }
 
+// logic to create (tmp) prefix directory
+func (obj *Main) createPrefix(hostname string) (string, error) {
+	var prefix string
+	var err error
+	if obj.TmpPrefix {
+		// create temporary prefix
+		prefix, err = ioutil.TempDir("", fmt.Sprintf("%s-%s-", obj.Program, hostname))
+		if err != nil {
+			return prefix, fmt.Errorf("can't create tmp-prefix: %s", err)
+		}
+		log.Println("Main: Warning: Working prefix directory is temporary!")
+	} else {
```

Comment:
Please try and inline the code flow, eg: https://github.com/golang/go/wiki/CodeReviewComments#indent-error-flow

Reply:
makes sense